### DISCUSSION TITLE
Share secret for usage in workspace setup?

### DISCUSSION THREAD

**Discussion by kohlerm** [coder/coder#6305] [General] [answered]
Hi,
I am using my own coder installation on a kubernetes cluster. 
 When the user creates the workspace, I need a secret specified by the user (somehow passed to the coder CLI?), to set things up. 
An environment variable available for the startup_script would work, I guess.
Is there a simple way to do this?

**Comment by matifali** [comment, marked as answer]
I guess this is the correct way. You can take the secret as input from the user as a terraform variable.
